src/models/TaskModels.ts

```diff
@@ -118,10 +125,15 @@ export class YamlConfig {
         if(config.certificates != undefined){
             this.certificates = this.parseParameters(config.certificates, ParamType.cert) as CertificateMetadata | null;
         }
-        if(config.keyVaultReferenceIdentity != undefined) {
-            this.keyVaultReferenceIdentityType = ManagedIdentityType.UserAssigned;
-            this.keyVaultReferenceIdentity = config.keyVaultReferenceIdentity;
+        if(config.keyVaultReferenceIdentity != undefined || config.keyVaultReferenceIdentityType != undefined) {
+            this.keyVaultReferenceIdentityType = config.keyVaultReferenceIdentity ? ManagedIdentityTypeForAPI.UserAssigned : ManagedIdentityTypeForAPI.SystemAssigned;
+            this.keyVaultReferenceIdentity = config.keyVaultReferenceIdentity ?? null;
+        }
+
+        if(config.referenceIdentities != undefined) {
+            this.getReferenceIdentities(config.referenceIdentities as {[key: string]: string}[]);
```

Comment:
Dont we need to handle the backcompat here?

Reply:
we did right ? we did in ado also and added uts also.